diff --git a/Python/remote_debug.h b/Python/remote_debug.h
--- a/Python/remote_debug.h
+++ b/Python/remote_debug.h
@@ -124,7 +124,7 @@ get_page_size(void) {
         GetSystemInfo(&si);
         page_size = si.dwPageSize;
 #else
-        page_size = (size_t)getpagesize();
+        page_size = (size_t)sysconf(_SC_PAGESIZE);
 #endif
     }
     return page_size;
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -15360,7 +15360,7 @@
 
 
 have_libffi=missing
-if test "x$ac_sys_system" = xDarwin
+if test "x$ac_sys_system" = xDarwin && test "x${PYTHON_IOS_EXTERNAL_LIBFFI-}" != xyes
 then :
 
   save_CFLAGS=$CFLAGS
